ansible_base/settings/dynamic_settings.py

```diff
@@ -66,3 +75,34 @@
             )
         }
     )
+
+if ANSIBLE_BASE_FEATURES.get(OAUTH2_PROVIDER_SETTING_STRING, False):  # noqa: F821
+    if 'oauth2_provider' not in INSTALLED_APPS:  # noqa: F821
+        INSTALLED_APPS.append('oauth2_provider')  # noqa: F821
+
+    try:
+        OAUTH2_PROVIDER  # noqa: F821
+    except NameError:
+        OAUTH2_PROVIDER = {}
+
+    if 'ACCESS_TOKEN_EXPIRE_SECONDS' not in OAUTH2_PROVIDER:
+        OAUTH2_PROVIDER['ACCESS_TOKEN_EXPIRE_SECONDS'] = 31536000000
+    if 'AUTHORIZATION_CODE_EXPIRE_SECONDS' not in OAUTH2_PROVIDER:
+        OAUTH2_PROVIDER['AUTHORIZATION_CODE_EXPIRE_SECONDS'] = 600
+    if 'REFRESH_TOKEN_EXPIRE_SECONDS' not in OAUTH2_PROVIDER:
+        OAUTH2_PROVIDER['REFRESH_TOKEN_EXPIRE_SECONDS'] = 2628000
+
+    OAUTH2_PROVIDER['APPLICATION_MODEL'] = 'ansible_base.OAuth2Application'
+    OAUTH2_PROVIDER['ACCESS_TOKEN_MODEL'] = 'ansible_base.OAuth2AccessToken'
+
+    oauth2_authentication_class = 'ansible_base.authentication.oauth2.LoggedOAuth2Authentication'
+    if 'DEFAULT_AUTHENTICATION_CLASSES' not in REST_FRAMEWORK:  # noqa: F821
+        REST_FRAMEWORK['DEFAULT_AUTHENTICATION_CLASSES'] = []  # noqa: F821
+    if oauth2_authentication_class not in REST_FRAMEWORK['DEFAULT_AUTHENTICATION_CLASSES']:  # noqa: F821
+        REST_FRAMEWORK['DEFAULT_AUTHENTICATION_CLASSES'].insert(0, oauth2_authentication_class)  # noqa: F821
+
+# These have to be defined for the migration to function
+OAUTH2_PROVIDER_APPLICATION_MODEL = 'ansible_base.OAuth2Application'
+OAUTH2_PROVIDER_ACCESS_TOKEN_MODEL = 'ansible_base.OAuth2AccessToken'
+OAUTH2_PROVIDER_REFRESH_TOKEN_MODEL = "ansible_base.OAuth2RefreshToken"
+OAUTH2_PROVIDER_ID_TOKEN_MODEL = "ansible_base.OAuth2IDToken"
```

Comment:
Nice to see this. In https://github.com/ansible/django-ansible-base/pull/75 I've also tried to add some settings that should be set unconditionally. It makes sense for this to become a thing.